aten/src/ATen/native/mkl/SpectralOps.cpp

```diff
@@ -556,6 +556,10 @@ Tensor _fft_r2c_mkl(const Tensor& self, IntArrayRef dim, int64_t normalization,
 // n-dimensional complex to complex FFT/IFFT
 Tensor _fft_c2c_mkl(const Tensor& self, IntArrayRef dim, int64_t normalization, bool forward) {
   TORCH_CHECK(self.is_complex());
+  if (dim.empty()) {
```

Comment:
Don't you need to add similar to PocketFFT implementation on line 308 (otherwise I assume it will fail on M1 Mac)